tests/puppeteer/campaigns/functional/BO/customers/addresses/02_CRUDAddressInBO.js

```diff
@@ -0,0 +1,153 @@
+require('module-alias/register');
+// Using chai
+const {expect} = require('chai');
+const helper = require('@utils/helpers');
+const loginCommon = require('@commonTests/loginBO');
+// Importing pages
+const BOBasePage = require('@pages/BO/BObasePage');
+const LoginPage = require('@pages/BO/login');
+const DashboardPage = require('@pages/BO/dashboard');
+const AddressesPage = require('@pages/BO/customers/addresses');
+const AddAddressPage = require('@pages/BO/customers/addresses/add');
+// Importing data
+const AddressFaker = require('@data/faker/address');
+
+let browser;
+let page;
+let numberOfAddresses = 0;
+const createAddressData = new AddressFaker({email: 'pub@prestashop.com'});
+const editAddressData = new AddressFaker({email: ' '});
```

Comment:
The email is not an error ?

Reply:
We cannot edit the email address in the form, I chose to set space and test if the email === " " then don't write in the input -> https://github.com/PrestaShop/PrestaShop/pull/17086/files#diff-9974230dc5e3bbd4ca6c15eccda0274eR39

Reply:
Could you use `email: null` instead? and test for !== null value?

Reply:
No, the test doesn't pass with the null value, because when the email is empty a fake value will be generated.

Reply:
This is why you need to update https://github.com/PrestaShop/PrestaShop/pull/17086/files#diff-6c35bac677390a77fe6f0fd18fa26465R7
using a fake `' '` is a bad practice :/